src/bilayers/build/parse/cellprofiler_plugin_template.py.j2

```diff
@@ -0,0 +1,655 @@
+                    {% set first_image_input = inputs.values() | selectattr('type', 'equalto', 'image') | list %}
+                    {% if first_image_input %}
+                    {{ output_conf.get('name') }}_data = numpy.zeros_like({{ first_image_input[0].get('name') }}_data)
+                    {% else %} # Prevent Docker crashing
+                    {{ output_conf.get('name') }}_data = numpy.zeros((256, 256), dtype=numpy.uint8)
+                    {% endif %}
+                {% elif output_conf.get('type') == 'array' %}
+                {{ output_conf.get('name') }}_output_path = os.path.join(temp_output_dir, f"{unique_name}_{{ output_conf.get('name') }}.npy")
+                if os.path.exists({{ output_conf.get('name') }}_output_path):
+                    {{ output_conf.get('name') }}_data = numpy.load({{ output_conf.get('name') }}_output_path)
+                {% elif output_conf.get('type') == 'measurement' %}
+                {{ output_conf.get('name') }}_output_path = os.path.join(temp_output_dir, f"{unique_name}_{{ output_conf.get('name') }}.csv")
+                if os.path.exists({{ output_conf.get('name') }}_output_path):
+                    {{ output_conf.get('name') }}_data = {{ output_conf.get('name') }}_output_path
+                {% endif %}
+                {% endfor %}
+            except FileNotFoundError as ex:
+                LOGGER.error(f"Algorithm output files not found: {ex}")
+                raise FileNotFoundError(f"I'm sorry, the {{ docker_image.get('name') }} algorithm seems to have crashed and I'm not sure why, since it's running in its own container. Expected output files were not created. Please check the algorithm parameters and try again.")
+            except Exception as ex:
+                LOGGER.error(f"Error loading algorithm outputs: {ex}")
+                raise RuntimeError(f"Failed to load algorithm outputs: {ex}")
+
+        finally:
+            # Cleanup temporary files
+            if self.clean_temp_files.value:
+                try:
+                    shutil.rmtree(temp_dir)
+                except OSError as ex:
+                    LOGGER.warning(f"Unable to delete temporary directory {temp_dir}: {ex}")
+            else:
+                LOGGER.info(f"Preserving temporary files for debugging at: {temp_dir}")
+                LOGGER.info(f"You can manually inspect the temporary files at: {temp_dir}")
+
+        # Process outputs and add to workspace
+        self._process_outputs(workspace)
+
+    def _process_outputs(self, workspace):
+        """Process algorithm outputs and add them to the workspace."""
+        # Handle outputs programmatically for all modules
+        LOGGER.info(f"Processing custom module outputs (category: {{ computed_category }})")
+        {% for key, output_conf in outputs.items() %}
+        {% if output_conf.get('type') == 'image' %}
+        {% if computed_category == "Image Segmentation" %}
```

Comment:
I think we want to do this based on subtype of label (or not), right? Because a segmenter might make an image AND an object, for example